docs/source/core_specification/schema_types.rst

```diff
@@ -1,3 +1,282 @@
+- :code:`out_name: String!` is a GraphQL argument. The :code:`!` indicates that it must not be null.
+- :code:`on FIELD` defines where the directive can be located. According to the definition, this
+  directive can only be located next to fields. The compiler might have additional restrictions
+  for where a query can be located. See :doc:`query directives <query_directives>` for more info.
+
+Scalars
+-------
+
+The compiler uses the built-in GraphQL
+`scalar types <https://graphql.org/learn/schema/#scalar-types>`__ as well as three custom scalars:
+
+- :code:`DateTime` represents timezone-aware second-accuracy timestamps.
+- :code:`Date` represents day-accuracy date objects.
+- :code:`Decimal` is an arbitrary-precision decimal number object useful for representing values
+  that should never be rounded, such as currency amounts.
+
+Query Operation
+---------------
+
+GraphQL allows for three operation types *query*, *mutation* and *subscription*. The compiler
+only allows query operation types as shown in the code snippet below:
+
+.. code::
+
+    schema {
+        query: RootSchemaQuery
+    }
+
+A query may begin in any of the **root vertex types** specified by the :code:`RootSchemaQuery`
+object:
+
+.. code::
+
+    type RootSchemaQuery {
+        Animal: [Animal]
+        Entity: [Entity]
+        Food: [Food]
+        Species: [Species]
+        Toy: [Toy]
+    }s
+
+Inheritance
+-----------
+
+The compiler uses GraphQL interfaces and GraphQL unions in representing the inheritance structure
+of the underlying schema. You may ignore this section if are compiling to a language with no
+inheritance structure, (e.g. SQL).
+
+Interfaces
+~~~~~~~~~~
+
+GraphQL interfaces represent the abstract vertices of the underlying database.
+
+.. code::
+
+    interface Entity {
+        name: String
+        alias: [String]
+        in_Entity_Related: [Entity]
+        out_Entity_Related: [Entity]
+    }
+
+Abstract inheritance is modeled through interface implementation.
+
+.. code::
+
+    type Food implements Entity {
+        name: String
+        alias: [String]
+        in_Entity_Related: [Entity]
+        out_Entity_Related: [Entity]
+        in_Species_Eats: [Species]
+    }
+
+Unions and :code:`type_equivalence_hints`
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+GraphQL does not support a notion of concrete inheritance. In other words, GraphQL objects cannot
+inherit from other objects. However, for certain query dialects, the compiler needs concrete
+inheritance information to emit the right query.
+
+To model concrete inheritance. we use GraphQL unions that encompass an object's
+subclasses and a :code:`type_equivalence_hints` parameter to signify that object is equivalent to
+the GraphQL union.
+
+For example, suppose :code:`Food` and :code:`Species` are concrete types and :code:`Food` is a
+superclass of :code:`Species` in an OrientDB schema. Then the GraphQL schema generation function
+would generate an union class in the schema
```

Comment:
Missing period.

Reply:
The sentence is continued below. Let me know if you don't like this.